zebra-state/src/request.rs

```diff
@@ -81,6 +81,8 @@ pub struct PreparedBlock {
     // TODO: add these parameters when we can compute anchors.
     // sprout_anchor: sprout::tree::Root,
     // sapling_anchor: sapling::tree::Root,
+    /// Storage for all the utxos related to this block.
+    pub block_utxos: HashMap<transparent::OutPoint, transparent::Utxo>,
```

Comment:
We need a place to keep track of all the utxos for the block because we are adding them in stages.